src/components/Experience.ts

```diff
@@ -29,11 +29,54 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends GeneralCo
      *
      * @param pokemon   The pokemon to give experience to.
      * @param experience   The amount of experience to give.
+     * @returns True if the pokemon leveled up, false otherwise.
      */
-    public gainExperience(pokemon: IPokemon, experience: number): void {
+    public gainExperience(pokemon: IPokemon, experience: number): boolean {
         pokemon.experience += experience;
         if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
             this.levelup(pokemon);
+            return true;
         }
+        return false;
+    }
+
+     /**
+      * Processes experience gain for one or more pokemon
+      *
+      * @param battleInfo   Information about the current battle.
+      * @param partyIsWipedText   Text to be displayed when the opposing pokemon faints.
+      *
+      */
+    public processBattleExperience(battleInfo: IBattleInfo, onComplete: () => void): void {
+        const experienceToGain =  this.gameStarter.equations.experienceGained(
+            battleInfo.teams.player, battleInfo.teams.opponent);
+
+        // BattleInfo should keep track of all pokemon the player has sent out in the battle and process
+        // them one by one if they have not fainted to allow for shared exp. The same logic will apply
+        // for an exp share where the pokemon holding the exp share is always considered to have been "sent out"
+        // The code below will be applied to each sent out in the battle
```

Comment:
Great comments, but I'm not a fan of putting comments in code. Code frequently changes and makes the comments out of date. Please move to the relevant GH issues (or create them if they don't yet exist).